Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselPanel.cs

```diff
@@ -0,0 +1,455 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    public class CarouselPanel : Panel
```

Comment:
It won't work.
This CarouselPanel is needed because you want an ItemsControl.
That's why, when you create an object inheriting from ItemsControl, if you follow the guidelines and how it's implemented in object like ListView, you should create an ItemsPanel.
For Carousel , the problem is that the only ItemsTemplate that works is the CarouselPanel. 
And in the other side, the only ItemsControl that could work with a CarouselPanel is the Carousel control.
Maybe in a next release, we could try to resolved this design problem :)